SignalServiceKit/src/Messages/OWSMessageManager.m

```diff
@@ -1442,6 +1442,20 @@ - (void)finalizeIncomingMessage:(TSIncomingMessage *)incomingMessage
         return;
     }
 
+    // Loki:
+    // ========
+    if (envelope.type == SSKProtoEnvelopeTypeFriendRequest) {
+        if (thread.hasCurrentUserSentFriendRequest) {
+            [thread setFriendRequestStatus:TSThreadFriendRequestStatusFriends withTransaction:transaction];
+        } else {
+            [thread setFriendRequestStatus:TSThreadFriendRequestStatusRequestReceived withTransaction:transaction];
```

Comment:
We only want to set the status to `requestReceived` if we're not friends with the person.

Reply:
The only way we can receive an `SSKProtoEnvelopeTypeFriendRequest` envelope type is if we're not already friends, right?

Reply:
you are correct 👍